src/plugins/data_source/server/client/client_config.test.ts

```diff
@@ -26,4 +29,54 @@ describe('parseClientOptions', () => {
       })
     );
   });
+
+  test('test ssl config with verification mode set to none', () => {
```

Comment:
looks like we need another test case `test ssl config with verification mode set to certificate`, also curious what's the difference between mode `full` and `certificate`

Reply:
I will add a test for this case

Reply:
Added test for `certificate` mode that asserts that it overwrites `checkServerIdentity`